### PR TITLE
Replace individual data constants with a class to set them all

### PR DESCRIPTION
**Pull request by lachlanhardy**
This resolves DOC-448:
https://linear.app/buildkite/issue/DOC-448/automatically-bind-data-yaml-files-to-vars-like-this-in-erb-templates

Content YAML files are moved to `data/content` and we turn each one into a constant in Rails using the filename: so the contents of `agent_attributes.yml` is available in the running app as a hash called `AGENT_ATTRIBUTES`.

I added a few basic specs to make sure this works as expected.

### CODE REVIEW

File: pages/test_analytics/importing_json.md.erb [hunk 1]
```diff
@@ -189,7 +189,7 @@ end
     </tr>
   </thead>
   <tbody>
-    <% TEST_ANALYTICS_JSON['test_result']['fields'].each do |field| -%>
+    <% TEST_ANALYTICS_JSON_FIELDS_TEST_RESULT['fields'].each do |field| -%>
```
Comment (plaindocs) [inline pages/test_analytics/importing_json.md.erb:192]: How does this know to grab `test_result` thing? 
I reckon a more sustainable approach might just be to split the YAML file into three files, and refer to them as usual.

Comment (lachlanhardy) [reply pages/test_analytics/importing_json.md.erb:192]: It *is* three YAML files. The current method unifies them into one hash. This PR stops doing that in favour of the automatic constant assignation.

See app/models/test_analytics_json.rb  at https://github.com/buildkite/docs/pull/1688/files#diff-76488f8d80ee6eee40647959662a65050aafd40648338f56aca1059bf3235d42L1

Comment (plaindocs) [reply pages/test_analytics/importing_json.md.erb:192]: Oh yeah, got it. Thanks! :-) 

So I guess my question is whether it's more intuitive to unify them, or leave them as three separate files, and I'd lean towards the second (regardless of what the original config is). ;-)

**Review by ddbeck — APPROVED** [review]
Looks good to me (though it prompted another, non-blocking issue, noted in a comment). Thank you!

File: package.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
     "husky": "^8.0.0"
   },
   "scripts": {
-    "validate-agent-attributes-yaml": "npx --package ajv-cli ajv validate -s \"data/agent_attributes.schema.yaml\" -d \"data/agent_attributes.yaml\"",
-    "validate-environment-variables-yaml": "npx --package ajv-cli ajv validate -s \"data/environment_variables.schema.yaml\" -d \"data/environment_variables.yaml\"",
+    "validate-agent-attributes-yaml": "npx --package ajv-cli ajv validate -s \"data/content/agent_attributes.schema.yaml\" -d \"data/content/agent_attributes.yaml\"",
+    "validate-environment-variables-yaml": "npx --package ajv-cli ajv validate -s \"data/content/environment_variables.schema.yaml\" -d \"data/content/environment_variables.yaml\"",
```
[the remark this answers is not in the document]
Comment (lachlanhardy) [reply package.json:9]: Great! I thought the same

**Comment by plaindocs** [comment]
Cool!! I didn't get a chance to look at this, but I'm happy it happened! Thanks @lachlanhardy !

@olyism FYI also

**Pull request merged**